lib/blazer/sharing.rb

```diff
@@ -0,0 +1,36 @@
+module Blazer
+  class Sharing
+    attr_accessor :api_key, :path
+
+    def initialize(api_key: ENV.fetch('BLAZER_DOWNLOAD_API_KEY', nil), path: '/blazer_share')
+      @api_key = api_key
+      @path = path.sub(/\/$/, '') # Strip trailing /
+    end
+
+    def route_path
+      @route_path ||= "#{path}/:token/:query_id"
+    end
+
+    def to_controller
+      'blazer/queries#share'
+    end
+
+    def query_token(query_id)
+      Digest::SHA1.hexdigest("#{query_id}-#{api_key}")
```

Comment:
This should use an HMAC, which was specifically designed to address [the security issues with hashing simple concatenations of the key and message](https://en.wikipedia.org/wiki/HMAC#Design_principles). While we're at it we can use SHA-256 (future proof security) with base64 encoding (trim back the longer length compared with SHA-1).

```suggestion
      hmac = OpenSSL::HMAC.digest("sha256", api_key, query_id)
      Base64.urlsafe_encode64(hmac)
```

Reply:
IMO better alternative would be to add a column to the queries table that stores a randomly generated token, so revoking is possible on a per-query basis instead of all queries (by changing the API key). Could be implemented using [ActiveRecord::SecureToken](https://api.rubyonrails.org/classes/ActiveRecord/SecureToken/ClassMethods.html).

Reply:
I completely agree with your solution to use SecureToken instead of an API key. It is a lot better! 

I've implemented it in the latest commits. Thanks for suggesting it.